### PR TITLE
Added new command to manage init logic

### PR DESCRIPTION
**Pull request by sabre1041**
Controls the management of execution based on pod annotations mounted into a file that is fed as part of the downward API

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -12,7 +12,7 @@ ENV GOPATH=/opt/app-root/go \
 ADD . /opt/app-root/go/src/github.com/sabre1041/istio-pod-network-controller
 
 RUN yum repolist > /dev/null && \
-#    yum-config-manager --enable rhel-7-server-optional-rpms --enable rhel-7-server-extras-rpms && \
+    yum-config-manager --enable rhel-7-server-optional-rpms --enable rhel-7-server-extras-rpms && \
```
Comment (raffaelespazzoli) [inline Dockerfile]: why did you have to re-add this? centos does not need it.

File: cmd/istio-pod-network-controller/init/init.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	if err != nil {
+		logrus.Errorf("Error occurred waiting for pod annotation in file: %v", err)
+		os.Exit(1)
+	}
+
+	logrus.Printf("Completed Initialization Successfully")
+
+}
+
+func getDefaultFileNameProperty() string {
```
Comment (raffaelespazzoli) [inline cmd/istio-pod-network-controller/init/init.go]: this piece of code smells for me. if what you wanted to do was to be able to initialize the value either from a flag or an environment variable, that is already achieved by this code in main.go:
```func initConfig() {
	viper.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))

	// read in environment variables that match
	viper.AutomaticEnv()
}```

so if that was the objective, you shouldn't need these functions and you should use the format I used in the run command to initialize the flags

File: pkg/init/init.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+)
+
+const PodAnnotationsFileName = "/etc/podinfo/pod_annotations"
+const PodAnnotationsKeyName = "pod-network-controller.istio.io/status"
```
Comment (raffaelespazzoli) [inline pkg/init/init.go]: we may want to create a separate go file with all the constants, this type of refactoring would limit duplication of codes, see also: handler.go L22 for all the needed constans

**Comment by raffaelespazzoli** [comment]
I didn't see the changes to the heuristics logic to filter which pods the controller should initialize (I think the email explained the needed logic).
Also I didn't see the changes to the injection config map to add the new init container.
Also after the above changes the docs will need some update.

File: Dockerfile (hunk 1 above)
Comment (sabre1041) [reply Dockerfile]: Needed for RHEL

File: pkg/init/init.go (hunk 3 above)
Comment (sabre1041) [reply pkg/init/init.go]: Its a good idea. Let's add this as an enhancement afterwards

**Comment by raffaelespazzoli** [comment]
I think we can merge this.

**Pull request merged**